public/app/core/components/AccessControl/Permissions.tsx

```diff
@@ -91,12 +91,12 @@ export const Permissions = ({
 
   const onRemove = (item: ResourcePermission) => {
     let promise: Promise<void> | null = null;
-    if (item.userId) {
-      promise = setUserPermission(resource, resourceId, item.userId, EMPTY_PERMISSION);
-    } else if (item.teamId) {
-      promise = setTeamPermission(resource, resourceId, item.teamId, EMPTY_PERMISSION);
-    } else if (item.isServiceAccount && item.userId) {
-      promise = setUserPermission(resource, resourceId, item.userId, EMPTY_PERMISSION);
+    if (item.userUid) {
+      promise = setUserPermission(resource, resourceId, item.userUid, EMPTY_PERMISSION);
+    } else if (item.teamUid) {
+      promise = setTeamPermission(resource, resourceId, item.teamUid, EMPTY_PERMISSION);
+    } else if (item.isServiceAccount && item.userUid) {
+      promise = setUserPermission(resource, resourceId, item.userUid, EMPTY_PERMISSION);
```

Comment:
Nit: feels like we could simplify this a bit and not have a separate clause for the service account? Same goes for the code below.